gateway/dataimpl/server_v1/queryadminserver.go

```diff
@@ -239,16 +315,7 @@ func (s *QueryIndexAdminServer) CreatePrimaryIndex(
 
 	_, err := s.executeQuery(ctx, &in.BucketName, qs)
 	if err != nil {
-		if errors.Is(err, cbqueryx.ErrIndexExists) {
-			var name string
-			if in.Name == nil {
-				name = "#primary"
-			} else {
-				name = *in.Name
-			}
-			return nil, s.errorHandler.NewQueryIndexExistsStatus(err, name).Err()
-		}
-		return nil, s.errorHandler.NewGenericStatus(err).Err()
+		return nil, s.makeSuitableStatus(err, in.Name, &in.BucketName, in.ScopeName, in.CollectionName)
```

Comment:
I think that typically in other places we've not used utility functions and handled the errors specificly for individual calls.

Reply:
I wanted to avoid duplicating the error checking so I thought this seemed the best approach. How would you suggest going about it?

Reply:
If you look at kvserver we do just duplicate the error handling where we have to.

Reply:
What's the reason for favouring the duplication of the handling over using utility functions?

Reply:
The major reason why we took that approach is that it makes it trivial for someone looking at the kvserver.go file to understand what incoming requests can lead to what specific outgoing errors.  An example of this is that memcached has a significant number of different errors, but a lot of them are impossible on some operations, for instance a BadDelta error only could occur when doing an Increment/Decrement operation, but if we generically handled all memcached errors in a single handler it means that its impossible to know whether BadDelta is an error you can get from a CNG Upsert operation.  We made the decision that we would be explicit about the error handling in CNG to create a 'hard' boundary where the errors produced by CNG are strictly defined without needing to go through a very deep dive of the memcached code to understand.

Reply:
I see, I've removed the utility function